src/Core/Entity/OptionsBag.php

```diff
@@ -146,6 +146,15 @@ public function getOption($key)
         return array_key_exists($key, $this->optionsCollection) ? $this->optionsCollection[$key] : '';
     }
 
+    /**
+     * Returns the AppParamters Object
+     * @return AppParameters
+     */
+    public function appParameters()
+    {
+        return $this->appParameters;
```

Comment:
isn't it exposing too much? it does the work though

Reply:
No I think it is ok, and we don't need to instantiate the class again for one use case.

Reply:
Agree, definitively not that